vespa-enforcer-extensions/src/test/java/com/yahoo/vespa/maven/plugin/enforcer/EnforceDependenciesAllProjectsTest.java

```diff
@@ -0,0 +1,84 @@
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import static com.yahoo.vespa.maven.plugin.enforcer.EnforceDependenciesAllProjects.validateDependencies;
+import static com.yahoo.vespa.maven.plugin.enforcer.EnforceDependenciesAllProjects.writeDependencySpec;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+/**
+ * @author bjorncs
+ */
+class EnforceDependenciesAllProjectsTest {
+
+    @Test
+    void succeeds_dependencies_matches_spec() {
+        SortedSet<Dependency> dependencies = new TreeSet<>(Set.of(
+                Dependency.fromString("com.example:foo:1.2.3"),
+                Dependency.fromString("com.example:bar:2.3.4")));
+        Path specFile = Paths.get("src/test/resources/allowed-dependencies.txt");
+        assertDoesNotThrow(() -> validateDependencies(dependencies, specFile));
+    }
+
+    @Test
+    void fails_on_forbidden_dependency() {
+        SortedSet<Dependency> dependencies = new TreeSet<>(Set.of(
+                Dependency.fromString("com.example:foo:1.2.3"),
+                Dependency.fromString("com.example:bar:2.3.4"),
+                Dependency.fromString("com.example:foobar:3.4.5")));
+        Path specFile = Paths.get("src/test/resources/allowed-dependencies.txt");
+        var exception = assertThrows(EnforcerRuleException.class,
+                                     () -> validateDependencies(dependencies, specFile));
+        String expectedErrorMessage =
+                """
+                The dependency enforcer failed:
+                Forbidden dependencies:
+                 - com.example:foobar:3.4.5
+                Maven dependency validation failed. To update dependency spec run 'mvn enforcer:enforce -DdependencyEnforcer.writeSpec'""";
+        assertEquals(expectedErrorMessage, exception.getMessage());
+    }
+
+    @Test
+    void fails_on_missing_dependency() {
+        SortedSet<Dependency> dependencies = new TreeSet<>(Set.of(
+                Dependency.fromString("com.example:foo:1.2.3")));
+        Path specFile = Paths.get("src/test/resources/allowed-dependencies.txt");
+        var exception = assertThrows(EnforcerRuleException.class,
+                                     () -> validateDependencies(dependencies, specFile));
+        String expectedErrorMessage =
+                """
+                The dependency enforcer failed:
+                Removed dependencies:
+                 - com.example:bar:2.3.4
+                Maven dependency validation failed. To update dependency spec run 'mvn enforcer:enforce -DdependencyEnforcer.writeSpec'""";
+        assertEquals(expectedErrorMessage, exception.getMessage());
+    }
+
```

Comment:
Do you need to test that version has changed.
 Fx com.example:bar:2.3.4 vs com.example:bar:2.3.5 ?

Reply:
Different versions are handled as separate artifacts